src/porepy/compositional/compositional_mixins.py

```diff
@@ -1222,106 +1217,102 @@ def dependencies_of_phase_properties(
-        )
+            - :attr:`density_of_phase`
+            - :attr:`specific_enthalpy_of_phase`
+            - :attr:`viscosity_of_phase`
+            - :attr:`thermal_conductivity_of_phase`
+            - :attr:`fugacity_coefficient`
 
-    def fluid_density(self, domains: pp.SubdomainsOrBoundaries) -> pp.ad.Operator:
-        """Returns an operaror by calling phase saturations and densities, and
-        performing a weighed sum."""
+            returning functions which raise an error when called, alerting the user
+            that some some mixin is missing to provide respective functions as the
+            implementation of some heuristic law.
 
-        op = pp.ad.sum_operator_list(
-            [
-                phase.saturation(domains) * phase.density(domains)
-                for phase in self.fluid_mixture.phases
-            ],
-            "fluid-mixture-density",
-        )
-        return op
-
-    def fluid_specific_volume(
-        self, domains: pp.SubdomainsOrBoundaries
-    ) -> pp.ad.Operator:
-        """Returns the reciprocal of whatever :meth:`fluid_density` returns."""
-        op = self.fluid_density(domains) ** pp.ad.Scalar(-1)
-        op.set_name("fluid-mixture-specific-volume")
-        return op
-
-    def fluid_specific_enthalpy(
-        self, domains: pp.SubdomainsOrBoundaries
-    ) -> pp.ad.Operator:
-        """Returns an operaror by calling phase fractions and specific enthalpies, and
-        performing a weighed sum.
+        Example:
 
-        Raises:
-            CompositeModellingError: If :attr:`equilibrium_type` is None, and hence no
-                phase fractions were introduced by :class:`CompositionalVariables`.
-                The consistent definition of the specific molar enthalpy of a fluid
-                mixture always depends on phase fractions.
+            Let's assume a general mixture with multiple components and phases, which
+            uses the :class:`CompositionalVariables`, and either
+            :attr:`_MixtureDOFHandler._has_unified_equilibrium` or
+            :attr:`_MixtureDOFHandler._has_equilibrium` defined. I.e., the
+            fluid and phase properties are not characterized by some mixed-in, heuristic
+            law, but by a local phase equilibrium system.
+            The above mentioned logic can then be translated into code the following
+            way:
 
-        """
+            .. code-block:: python
 
-        if not self._has_equilibrium:
-            raise CompositionalModellingError(
-                "Attempting to define the (specific) fluid mixture enthalpy as sum of"
-                + " phase enthalpies weighed with phase fractions, even though no"
-                + " equilibrium conditions defined. Per default, phase fractions are"
-                + " only created when an equilibrium formulation is used in the model."
-            )
+                class MyMixin:
```

Comment:
I think we should aim at migrating some of these examples to a tutorial

Reply:
Yes, this will be part of the coming tutorial on tracer flow, which will be part of #1236